### ISSUE TITLE
woff2 fonts missing

### ISSUE THREAD

**Issue by radinamatic** [learningequality/ka-lite#4911]
## Summary

I cannot find the issue for `glyphicons-halflings-regular.woff2` not displaying, (it's used on contentrec page), but today I saw that `KaTeX_Main-Regular.woff2` is failing too. Could this be a "general" issue with `woff2` format?   

## Branch or installer method

Alpha3 deb installer on Ubuntu 0.15, both FF & Chromium affected.

## Traceback or relevant snippet from server.log

```
[INFO] [2016-02-24 18:24:45,561] django.request: HTTP Request /api/assessment_item/x5002ff6c75655ad3/ - Response: 200
[INFO] [2016-02-24 18:24:46,060] cherrypy.access.140462054089168: 127.0.0.1 - - [24/Feb/2016:18:24:46] "GET /static/fonts/KaTeX_Main-Regular.woff2 HTTP/1.1" 404 1394 "http://127.0.0.1:8008/learn/khan/science/physics/two-dimensional-motion/skill-check-for-two-dimensional-motion/skill-check-two-dimensional-motion/" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Ubuntu Chromium/48.0.2564.116 Chrome/48.0.2564.116 Safari/537.36"
[INFO] [2016-02-24 18:24:46,116] cherrypy.access.140462054089168: 127.0.0.1 - - [24/Feb/2016:18:24:46] "GET /static/fonts/KaTeX_Main-Regular.woff HTTP/1.1" 200 37620 "http://127.0.0.1:8008/learn/khan/science/physics/two-dimensional-motion/skill-check-for-two-dimensional-motion/skill-check-two-dimensional-motion/" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Ubuntu Chromium/48.0.2564.116 Chrome/48.0.2564.116 Safari/537.36"
```
```
[INFO] [2016-02-24 18:24:41,730] django.request: HTTP Request /securesync/api/user/status/ - Response: 200
[INFO] [2016-02-24 18:24:41,822] cherrypy.access.140462054089168: 127.0.0.1 - - [24/Feb/2016:18:24:41] "GET /static/fonts/glyphicons-halflings-regular.woff2 HTTP/1.1" 404 1414 "http://127.0.0.1:8008/learn/khan/science/physics/two-dimensional-motion/skill-check-for-two-dimensional-motion/skill-check-two-dimensional-motion/" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Ubuntu Chromium/48.0.2564.116 Chrome/48.0.2564.116 Safari/537.36"
[INFO] [2016-02-24 18:24:41,845] django.request: HTTP Request /api/topic_tree/khan - Response: 200
```

## How to reproduce

1. Check the contentrec page & and the `/science/physics/two-dimensional-motion/skill-check-for-two-dimensional-motion/skill-check-two-dimensional-motion/` exercise.

## Screenshots

![katex2](https://cloud.githubusercontent.com/assets/1457929/13294965/c1511c7e-db26-11e5-855f-3d240af11c4f.png)

![katex](https://cloud.githubusercontent.com/assets/1457929/13294972/c72e4130-db26-11e5-81f0-2f22631c14a2.png)

**Comment by rtibbles** [comment]
More likely it is an issue with the path being set for the fonts.

Could you check the static folder and see if the fonts appear in those places, or if they are actually somewhere else?

**Comment by radinamatic** [comment]
Sorry, I haven't mentioned that earlier @rtibbles, static folder was the first place I looked, and no `woff2` fonts there, just a long line of "normal" `woff` fonts.

**Comment by rtibbles** [comment]
Guess we'd better get the woff2 files in there too. Will investigate.

**Comment by radinamatic** [comment]
Testing this on 0.16 branch with OSX gives no 404 response, however it displays differently across browsers:

FF

![katex3](https://cloud.githubusercontent.com/assets/1457929/13307319/b8c43606-db6a-11e5-87d7-c68c0926fef0.png)

Chrome (no idea if this is what it should be like)

![katex5](https://cloud.githubusercontent.com/assets/1457929/13307335/cd76a3e0-db6a-11e5-9a43-5e7456e28c16.png)

Safari (first time I'm opening this ;)

![katex4](https://cloud.githubusercontent.com/assets/1457929/13307375/0f2cc74c-db6b-11e5-83b5-125eacb13de3.png)

**Comment by rtibbles** [comment]
Very odd, the woff2 files are there in the source, not quite sure why they wouldn't be being copied over into the static folder at build time..

**Comment by radinamatic** [comment]
I'm testing upgrade scenarios on OSX atm, and seeing that woff2 fonts were missing on 0.15 too. So this may be going on for a while now...

**Comment by rtibbles** [comment]
Very odd, as the woff2 font files are also in the static folder in 0.15.x

Is there something about the file extension `.woff2` that might be causing an issue on the systems on which you are testing? (I can't think of one, but just don't understand why `collectstatic` is seemingly not doing its job).

**Comment by radinamatic** [comment]
Well, unless there is gremlin that s***** with all of my 4 testing computers and 14 virtual machines (here and in the office) where this error has been consistent for a while, I can't think of a thing... :grinning:

**Comment by radinamatic** [comment]
I was checking [this](https://developer.mozilla.org/en-US/docs/Web/Guide/WOFF#Browser_compatibility) @rtibbles, but all my FF browsers have the `gfx.downloadable_fonts.woff2.enabled` set to `true`. This is also an issue with deployments who must use IE, as it is not supported.

Funny thing is on my dev repo, I don't see error in the console, but they still don't display (FF 44):

![woff2](https://cloud.githubusercontent.com/assets/1457929/13409864/21ac339e-df35-11e5-8328-b027ec2a5eb4.png)

**Comment by rtibbles** [comment]
Could you try accessing the same exercise on the Khan Academy website in the same environment, and seeing if they render?

**Comment by radinamatic** [comment]
Now, why didn't I think of that... :disappointed: 

I can't see a single font loaded on their site, they must be serving it some other way???

https://www.khanacademy.org/science/physics/two-dimensional-motion/skill-check-for-two-dimensional-motion/e/skill-check-two-dimensional-motion

![kawoff2](https://cloud.githubusercontent.com/assets/1457929/13410162/8a7c2c48-df36-11e5-9146-802b3c88095e.png)

**Comment by rtibbles** [comment]
Sneaky buggers.

**Comment by rtibbles** [comment]
Looks like the font in question is the 'FontAwesome' family - will double check that this is included in our fonts.

**Comment by rtibbles** [comment]
Have also replicated this locally, @radinamatic - I think it's a matter of us not having all the fonts we need.

**Comment by rtibbles** [comment]
I have it loading the 'FontAwesome' family now, but this does not seem to resolve the issue, it just makes the missing font squares smaller!

![image](https://cloud.githubusercontent.com/assets/1680573/13441840/2899d750-dfae-11e5-9e43-79cbc95cd8fa.png)

**Comment by rtibbles** [comment]
As this seems to be limited to very few exercises, I am not going to mark this as a release blocker.

**Comment by radinamatic** [comment]
Do you have another fancy script to find out how many exercises make use of woff2 fonts? :grinning:

**Comment by rtibbles** [comment]
Well, this particular issue is *not* related to woff2. Also, most of the woff2 fonts have fallbacks to non-woff2 type.

Are there any situations where the woff2 is actually causing a problem? The issue with the exercise above is just due to a font being missing _in toto_.

**Comment by radinamatic** [comment]
The bigger problem is the `glyphicons-halflings-regular.woff2` used on contentrec page, we should find the replacement.

**Comment by rtibbles** [comment]
Yeah, replicating that locally too. Will fix that.

**Comment by rtibbles** [comment]
Is this the issue you are seeing, or are the arrows and check mark missing too?

![image](https://cloud.githubusercontent.com/assets/1680573/13442424/fb473b0a-dfb0-11e5-8b6b-48634aa5938d.png)

**Comment by radinamatic** [comment]
That's the one, it is supposed to be a pencil icon. 
I can see arrow and check mark fine.

**Comment by rtibbles** [comment]
OK - the pencil icon is actually an icomoon icon. The arrow and check mark are glyphicons.

The issue is that glyphicon classes have been applied to an icomoon icon, and there is some sort of mismatch. I am having trouble understanding how it ever worked.

**Comment by radinamatic** [comment]
It is working on 0.15, so whatever we did in the last 3 months, must have given birth to the gremlin that's eating them.

**Comment by rtibbles** [comment]
Seems to be the addition of the 'glyphicon' class that is causing the issue.

The diff where this happened is here: https://github.com/learningequality/ka-lite/commit/c5313db637cc8921690157a3a88e83797e75330d

Look familiar? :smile:

**Comment by radinamatic** [comment]
![Image](http://justjennrecipes.com/wp-content/uploads/2013/03/picard-facepalm-cookies.jpg)

**Comment by radinamatic** [comment]
I was convinced that commit was merged in 0.15... :flushed:

**Comment by rtibbles** [comment]
Fixed in #4941.

@radinamatic can you open a new issue for the exercise rendering font problems?
